util.grace

```diff
@@ -396,31 +396,38 @@ method execDir {
     execDirCache
 }
 
-method file(name) on(origin) orPath(pathString) otherwise(action) {
+method files(names) on(origin) orPath(pathString) otherwise(action) {
     def locations = filePath.split(pathString)
     locations.addFirst(origin)
     if (locations.contains(outDir).not) then { locations.addFirst(outDir) }
         // it's important that outDir is first; if not, the compiler
         // might compile an imported file, but then be unable to find the
         // code that it just generated.
     if (locations.contains(execDir).not) then { locations.addLast(execDir) }
-    def candidate = name.copy
-    def originalDir = name.directory
-    if ((originalDir.size > 0) && {originalDir.first == "/"}) then {
-        if (candidate.exists) then {
-            return candidate
-        } else {
-            return action.apply ""
+    def candidates = sequence << names.map{n -> n.copy}
+    def originalDirs = sequence << names.map{n -> n.directory}
+    // This assumes that either everything in "names" is an absolute path, or nothing is
+    if ((names.size > 0) && {originalDirs.at(1).size > 0} && {originalDirs.at(1).first == "/"}) then {
```

Comment:
I don't see a need for either of the first two conjuncts.  If `names` is empty, the loop won't execute; `originalDirs` was populated by mapping `directory`so it can't be empty.  (In any case, `.isEmpty` is clearer and sometimes faster than `.size == 0`).

Reply:
Yes the `names.size` thing was pointless, `originalDirs.at(1).size > 0` is needed, since it will probably fail if `names` = `["foo.grace"]`,  causing the subsequent test, `originalDirs.at(1).first == "/"` to give an error.